Assets/Scripts/Player/PlayerUi.cs

```diff
@@ -28,13 +27,25 @@ void Awake()
     void Start()
     {
         mainCameraTransform = Camera.main.transform;
-        inventory = InventoryManager.Singleton;
         messageUI = GetComponent<MessageUI>();
     }
 
     public void SetActiveUI(bool value)
     {
-            ui.SetActive(value);
+        ui.SetActive(value);
+    }
+
+    private void FixedUpdate()
+    {
+        if (ui.activeInHierarchy)
+        {
+            if (!playerStateController.Select)
```

Comment:
Would be nice to have a similar comment as this piece of code had before the refactorization :)
```suggestion
        {
            // If the select button is no longer pressed, handle the selection
            if (!playerStateController.Select)
```